sn_node/src/put_validation.rs

```diff
@@ -385,29 +389,53 @@ impl Node {
         Ok(CmdOk::StoredSuccessfully)
     }
 
-    /// Gets CashNotes out of a Payment, this includes network verifications of the Transfer
+    /// Gets CashNotes out of a Payment, this includes network verifications of the Transfer.
+    /// Return the CashNotes corresponding to our wallet and the ones corresponding to network royalties payment.
     async fn cash_notes_from_payment(
         &self,
         payment: &Vec<Transfer>,
         wallet: &LocalWallet,
         pretty_key: PrettyPrintRecordKey,
-    ) -> Result<(Vec<CashNote>, Transfer), ProtocolError> {
+    ) -> Result<(Vec<CashNote>, Vec<CashNote>), ProtocolError> {
+        let royalties_pk = *NETWORK_ROYALTIES_PK;
+        let mut cash_notes = vec![];
+        let mut royalties_cash_notes = vec![];
+
         for transfer in payment {
-            match self
-                .network
-                .verify_and_unpack_transfer(transfer, wallet)
-                .await
-            {
-                // transfer not for us
-                Err(ProtocolError::FailedToDecypherTransfer) => continue,
-                // transfer invalid
-                Err(e) => return Err(e),
-                // transfer ok
-                Ok(cash_notes) => return Ok((cash_notes, transfer.clone())),
-            };
+            match transfer {
+                Transfer::Encrypted(_) => match self
+                    .network
+                    .verify_and_unpack_transfer(transfer, wallet)
+                    .await
+                {
+                    // transfer not for us
+                    Err(ProtocolError::FailedToDecypherTransfer) => continue,
+                    // transfer invalid
+                    Err(e) => return Err(e),
+                    // transfer ok
+                    Ok(cn) => cash_notes = cn,
+                },
+                Transfer::NetworkRoyalties(cashnote_redemptions) => {
+                    if let Ok(cash_notes) = self
+                        .network
+                        .verify_cash_notes_redemptions(royalties_pk, cashnote_redemptions)
+                        .await
+                    {
+                        trace!(
+                            "{} network royalties payment cash notes found for record {pretty_key}",
+                            cash_notes.len()
```

Comment:
shall we log the sum of values of those cash_notes?
I think that will be more interested to be known ?